src/Core/ServiceProvider.php

```diff
@@ -44,25 +44,11 @@ abstract class ServiceProvider extends AbstractServiceProvider implements Bootab
     /**
      * Get the container.
      *
-     * This method's actual return type and documented return type differ
-     * because PHP 7.2 doesn't support return type narrowing.
-     *
      * @return \Cake\Core\ContainerInterface
      */
     public function getContainer(): ContainerInterface
     {
-        $container = parent::getContainer();
-
-        if (!($container instanceof ContainerInterface)) {
-            $message = sprintf(
-                'Unexpected container type. Expected `%s` got `%s` instead.',
-                ContainerInterface::class,
-                get_debug_type($container)
-            );
-            throw new RuntimeException($message);
-        }
-
-        return $container;
+        return parent::getContainer();
```

Comment:
How is this set if it's supposed to be cake's ContainerInterface?

Reply:
Because Cake's `ContainerInterface` extends League's `ServiceProviderInterface`.

Reply:
I mean it returns the parent. I assume we set the parent instance somewhere?

Reply:
We don't need to explicitly set it, it's handled automatically:
https://github.com/thephpleague/container/blob/82a57588c630663d2600f046753b23ab6dcda9b5/src/Container.php#L52
https://github.com/thephpleague/container/blob/82a57588c630663d2600f046753b23ab6dcda9b5/src/ServiceProvider/ServiceProviderAggregate.php#L32